src/ngx_pagespeed.cc

```diff
@@ -544,7 +544,8 @@ ngx_int_t copy_response_headers_to_ngx(
     }
 
     if (STR_EQ_LITERAL(name, "Cache-Control")) {
-      ps_set_cache_control(r, const_cast<char*>(value_gs.c_str()));
+      ps_set_cache_control(r, string_piece_to_pool_string(
+          r->pool, value_gs.c_str()));
```

Comment:
Ok, so the problem here is that we have some PSOL-allocated memory, and we need to keep that around until nginx is done with the request.  Two ways:

a) what your change does, and what the original code is trying to do: make a copy that's bound to the lifetime of the request, via a request pool.

b) what you're suggesting now: keep the PSOL memory around until we learn that the request has completed.

The advantage I'm seeing to (a) is that it's simpler and frees up memory (the rest of NgxBaseFetch) sooner, while the advantage of (b) is that it does a little less copying (the cache-control header value).

Saving copying <20 bytes doesn't seem worth rewriting this all?  So I'd lean towards (a) and just making the small fix.